### ISSUE TITLE
iOS Binding Library - Cannot find types declared

### ISSUE THREAD

**Issue by PureWeen** [xamarin/xamarin-macios#17541]
---
**Issue moved from dotnet/maui#13283**
- Please respond to @orwo1.

---

_From @orwo1 on Sunday, February 12, 2023 12:35:20 PM_

### Description

I've tried making an iOS binding library in VS for Mac.
I used Sharpie to create both definitions files, fixed what was wrong in them( NSArray<T> -> T[] ).
The project compiles.
I reference it in MAUI app and MAUI class library with condition:
       `<ItemGroup Condition="$(TargetFramework.Contains('-ios')) != false ">
	  <ProjectReference Include="..\iOSBinding1\iOSBinding1.csproj" />
	</ItemGroup>`
I Cannot access any type declared in the binding library, in my app/class library.

The Android Binding Library works fine under Visual Studio for Windows,
Under the reverse condition in the csproj.

### Steps to Reproduce

1. Create Binding Library wrapping a .framework with native reference and Sharpie's output in the:
    ApiDefinition.cs
    StructsAndEnums.cs
2. Reference the project in a Maui Class Library/Maui App.
3. Try to use any type declared in the binding library

### Link to public reproduction project repository

NA

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15, iOS 15.5, iOS 16.1

### Did you find any workaround?

No

**Comment by PureWeen** [comment]
---
**Issue moved from dotnet/maui#13283**
- Please respond to @orwo1.

---

_From @orwo1 on Sunday, February 12, 2023 12:37:29 PM_

More info:
That same .framework wrapped in Xamarin.Formsm iOS binding library, works in a Xamarin.Forms project.
I believe with the same Api and Enums output from Sharpie.

**Comment by PureWeen** [comment]
---
**Issue moved from dotnet/maui#13283**
- Please respond to @orwo1.

---

_From @orwo1 on Sunday, February 12, 2023 1:25:07 PM_

Using:
Visual Studio for Mac 17.4.4
XCode 14.2

**Comment by PureWeen** [comment]
---
**Issue moved from dotnet/maui#13283**
- Please respond to @msftbot[bot].

---

_From @msftbot[bot] on Monday, February 13, 2023 3:09:43 PM_

Hi @orwo1. We have added the "s/needs-repro" label to this issue, which indicates that we require steps and sample code to reproduce the issue before we can take further action. Please try to create a minimal sample project/solution or code samples which reproduce the issue, ideally as a GitHub repo that we can clone. See more details about creating repros here: https://github.com/dotnet/maui/blob/main/.github/repro.md

This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

**Comment by PureWeen** [comment]
---
**Issue moved from dotnet/maui#13283**
- Please respond to @orwo1.

---

_From @orwo1 on Tuesday, February 14, 2023 6:51:16 AM_

@jsuarezruiz Hi
I can't upload propriety .framework to github, in order to create a repro.
I've just created a solution with a basic MAUI class library,
an ios binding library,
and referenced it in the class library using the condition above.
There's nothing much to it other than that.

**Comment by rolfbjarne** [comment]
@orwo1 I believe this to be a duplicate of #10148 - does it work if you build the binding project separately, and then reference the resulting assemby (.dll) from the main project?

**Comment by orwo1** [comment]
> @orwo1 I believe this to be a duplicate of #10148 - does it work if you build the binding project separately, and then reference the resulting assemby (.dll) from the main project?

Hi @rolfbjarne 
Haven't tried that, will check this option and get back to you here.

**Comment by orwo1** [comment]
Hi @rolfbjarne 
Your suggestion seems to work, and by referencing the dll I was able
to access types defined in it.
I guess the next question is whether Microsoft knows of this issue?

**Comment by rolfbjarne** [comment]
> I guess the next question is whether Microsoft knows of this issue?

I'm not sure I understand what issue you're talking about?

**Comment by rolfbjarne** [comment]
Closing as duplicate of #10148.

**Comment by orwo1** [comment]
> > I guess the next question is whether Microsoft knows of this issue?
> 
> I'm not sure I understand what issue you're talking about?

The issue of a project reference not working the same way a dll file reference works (or vice versa).
This is not an issue?

**Comment by rolfbjarne** [comment]
> > > I guess the next question is whether Microsoft knows of this issue?
> > 
> > 
> > I'm not sure I understand what issue you're talking about?
> 
> The issue of a project reference not working the same way a dll file reference works (or vice versa). This is not an issue?

Yes, it's filed here: https://github.com/xamarin/xamarin-macios/issues/10148.

**Comment by orwo1** [comment]
> > > > I guess the next question is whether Microsoft knows of this issue?
> > > 
> > > 
> > > I'm not sure I understand what issue you're talking about?
> > 
> > 
> > The issue of a project reference not working the same way a dll file reference works (or vice versa). This is not an issue?
> 
> Yes, it's filed here: #10148.
Oh ok thanks for the help!
